### PR TITLE
Update the talk list

### PR DESCRIPTION
**Pull request by jonmeow**
The LLVM Developers' meeting agenda was [just published](https://discourse.llvm.org/t/announcing-the-2024-llvm-developers-meeting-program/81108), so add that.

### CODE REVIEW

**Comment by jonmeow** [comment]
Note, I double-checked C++Now, they're currently publishing their 2024 videos but geoffromer's talk still isn't published.
https://www.youtube.com/user/BoostCon/videos

**Review by zygoloid — APPROVED** [review]
LG, are we going to wait until Wednesday (after Chandler's talk) to merge?

**Comment by jonmeow** [comment]
I'm going to merge tomorrow, since then I can add a link to Chandler's slides.

File: README.md
Comment (josh11b) [inline README.md]: I believe this talk has now occurred.
```suggestion
-   [The Carbon Language: Road to 0.1](https://ndctechtown.com/agenda/the-carbon-language-road-to-01-0sqv/0526yb03a59),
    NDC {TechTown}, [slides](https://chandlerc.blog/slides/2024-ndc-techtown-carbon-road-to-0-dot-1)
```

**Review by josh11b — APPROVED** [review]
Now that the talk has occurred, can now merge this PR.

File: README.md
Comment (jonmeow) [reply README.md]: Thanks! I'm instead replacing the existing link.

**Comment by jonmeow** [comment]
Looks like https://chandlerc.blog/slides/index.html was fixed and now lists slides, so I've also added the CppNorth slides.

**Comment by jonmeow** [comment]
Note, this means the only finished talk we're missing video/slides for is the C++Now talk. Hopefully that's ready soon.

**Pull request merged**
